storage/src/test/java/org/apache/kafka/admin/RemoteTopicCrudTest.java

```diff
@@ -0,0 +1,698 @@
+        }
+        verifyRemoteLogTopicConfigs(topicConfig);
+    }
+
+
+    @ClusterTest
+    void testCreateRemoteTopicWithInheritedLocalRetentionTime() throws Exception {
+        // inherited local retention ms is 1000
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+            TopicConfig.RETENTION_MS_CONFIG, "1001"
+        );
+        try (var admin = cluster.admin()) {
+            admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig)));
+        }
+        verifyRemoteLogTopicConfigs(topicConfig);
+    }
+
+    @ClusterTest
+    void testCreateRemoteTopicWithInheritedLocalRetentionSize() throws Exception {
+        // inherited local retention bytes is 1024
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+            TopicConfig.RETENTION_BYTES_CONFIG, "1025"
+        );
+        try (var admin = cluster.admin()) {
+            admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig)));
+        }
+        verifyRemoteLogTopicConfigs(topicConfig);
+    }
+
+    @ClusterTest
+    void testCreateRemoteTopicWithInvalidRetentionTime() {
+        // inherited local retention ms is 1000
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+            TopicConfig.RETENTION_MS_CONFIG, "200"
+        );
+
+        try (var admin = cluster.admin()) {
+            assertThrowsExceptionIgnoringResult(() -> admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig))).all().get());
+        }
+    }
+
+    @ClusterTest
+    void testCreateRemoteTopicWithInvalidRetentionSize() {
+        // inherited local retention bytes is 1024
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+            TopicConfig.RETENTION_BYTES_CONFIG, "512"
+        );
+
+        try (var admin = cluster.admin()) {
+            assertThrowsExceptionIgnoringResult(() -> admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig))).all().get());
+        }
+    }
+   
+    @ClusterTest
+    void testCreateCompactedRemoteStorage() {
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+            TopicConfig.CLEANUP_POLICY_CONFIG, "compact"
+        );
+
+        try (var admin = cluster.admin()) {
+            assertThrowsExceptionIgnoringResult(() -> admin.createTopics(
+                List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig))).all().get());
+        }
+    }
+
+    // `remote.log.delete.on.disable` and `remote.log.copy.disable` only works in KRaft mode.
```

Comment:
Kafka supports only KRaft mode starting from 4.0, so this comment can be removed.